src/Mvc/Mvc.NewtonsoftJson/src/NewtonsoftJsonInputFormatter.cs

```diff
@@ -196,8 +196,15 @@ public virtual InputFormatterExceptionPolicy ExceptionPolicy
                 }
             }
 
-            if (!(exception is JsonException || exception is OverflowException))
+            if (!(exception is JsonException || exception is OverflowException || exception is FormatException))
```

Comment:
I dug around and `OverflowException` made the list because there was a test that broke if it wasn't allowed. Since it looks like an oversight, I don't think there's a compelling reason to put this behind a compat flag. The SystemTextJson one was a similar oversight. 

Thoughts?